001-config-overhaul.md

```diff
@@ -648,6 +688,138 @@ hiccups along the way.
 
 - [[suggestion] Trace configuration parameters resolution · Issue #3502 · hyperledger/iroha](https://github.com/hyperledger/iroha/issues/3502)
 
+### Proposal 9 - Remove Configuration Endpoints
+
+#### Objective
+
+To simplify and enhance the robustness of the Iroha system by eliminating the current configuration-related endpoints.
+
+#### Rationale
+
+1. **Superfluous Documentation Retrieval**: Fetching configuration field documentation through the API is unnecessary
+   and redundant. A well-maintained [configuration reference](#proposal-2---reference-before-implementation) should be
+   the single authoritative source for all such documentation.
+2. **Runtime Configuration Updates**: Allowing for run-time changes to the configuration can introduce complexity and
+   potential issues that undermine system robustness. A philosophy akin to systems like Elixir/Erlang emphasizes the
```

Comment:
How about explicitly splitting the node configuration into different parts where one includes just not related to the network stuff, i.e. 'local', like loglevel and could be changed in runtime. I guess it is pretty convenient to use this feature. Esp. considering how long it takes to restart a node if `fast` init mode is not set.

Reply:
AFAIK, after introducing [Snapshots](https://github.com/hyperledger/iroha/pull/3724) (enabled by default) start-up times do not suffer that much.

Having ability to set log level dynamically is a convenient feature, but it introduces a layer of complexity which, in my perspective, isn't worth its minor convenience.

I would vote for further optimisation of start-up times instead and making restart-ability in general a more available action to do.

Reply:
Though that would mean people who prefer not using snapshots for some reason are excluded from the set of 'dynamically changeable parameters enjoyers'.

Reply:
Currently - yup.

Maybe, if there is a strong need to have a temporary workaround, like it was with dynamic `LOG_LEVEL`, it makes sense to introduce a temporary unstable endpoint.

Reply:
While i agree with that currently changing `LogLevel` at runtime involve to much complexity and we can drop it for now. 

In the future if we find necessary hot-reload of configuration can be handled with [`ArcSwap`](https://docs.rs/arc-swap/latest/arc_swap/) which was specially was designed for this purpose.

I want to say that there is limit to how fast iroha could start (even with `FAST` mode and snapshots), because imo iroha is inherently statefull system despite common design approach where app is fully stateless and it can be reloaded freely.

And if we would take a look at statefull systems: [Postgres](https://pgpedia.info/p/pg_reload_conf.html), [redis](https://developer.redis.com/operate/redis-at-scale/talking-to-redis/configuring-a-redis-server/), ... Hot-reload of configuration is more common.

Reply:
That is a good point. I should probably reconsider removal of the endpoints. But instead, the endpoints should be re-designed to achieve the following:

- There will be a strict sub-set of parameters which will be exposed with `GET /configuration`. Others either are sensitive or meaningless to return (like storage path). It probably make sense to expose only those ones which could be updated with `POST /configuration`
- Format of updating should be fully compliant with the format in the config file/ENV.
- Documentation of the configuration should be clear enough to not confuse users: there is a config file, ENV, CLI arguments, dynamic HTTP configuration, and on-chain parameters.